### PR TITLE
Edge operators

### PR DESCRIPTION
**Pull request by martinRenou**
Add support for:
- selecting edges
- apply edge operators: chamfer and fillet

[Screencast from 2024-03-27 11-59-47.webm](https://github.com/jupytercad/jupytercad/assets/21197331/0c278c8c-07f1-49a1-beff-4e947dead643)

### CODE REVIEW

File: packages/base/src/mainview.tsx [hunk 1]
```diff
@@ -624,15 +651,11 @@ export class MainView extends React.Component<IProps, IStates> {
   private _onClick(e: MouseEvent) {
     const selection = this._pick();
     const selectedMeshesNames = new Set(
-      this._selectedMeshes.map(sel => sel.parent!.name)
+      this._selectedMeshes.map(sel => sel.name)
     );
 
     if (selection) {
-      // TODO Support selecting edges?
```
Comment (martinRenou) [inline packages/base/src/mainview.tsx:631]: done

**Comment by martinRenou** [comment]
please update snapshots

**Comment by martinRenou** [comment]
https://github.com/jupytercad/jupytercad/actions/runs/8454362416

File: ui-tests/tests/ui.spec.ts-snapshots/MultiSelect-test-jcad-linux.png
[the remark this answers is not in the document]
Comment (trungleduc) [reply ui-tests/tests/ui.spec.ts-snapshots/MultiSelect-test-jcad-linux.png:1]: I'm in line with the new behavior

File: packages/schema/src/schema/jcad.json [hunk 2]
```diff
@@ -43,6 +43,8 @@
         "Part::MultiFuse",
         "Part::MultiCommon",
         "Part::Extrusion",
+        "Edge::Chamfer",
```
Comment (trungleduc) [inline packages/schema/src/schema/jcad.json:46]: The `Part::` prefix comes from FreeCAD, and it also uses `Part::Chamfer` and `Part::Fillet` in a FCStd file, should we keep the same schema here?

File: packages/base/src/mainview.tsx [hunk 3]
```diff
@@ -125,9 +132,11 @@ export class MainView extends React.Component<IProps, IStates> {
       this
     );
 
-    if (this._raycaster.params.Line) {
-      this._raycaster.params.Line.threshold = 0.1;
-    }
+    // @ts-ignore Missing ThreeJS typing
```
Comment (trungleduc) [inline packages/base/src/mainview.tsx:135]: Upgrading `threejs` might help? It looks like the `Line2` property is added to the new interface
 https://github.com/three-types/three-ts-types/blob/056281e10e2b0aa0f6f2a126f9020b60e7d45759/types/three/src/core/Raycaster.d.ts#L42

Comment (martinRenou) [reply packages/base/src/mainview.tsx:135]: Indeed, I'm a bit worried about updating ThreeJS though, would it be fine for you doing this in a separate PR?

Comment (trungleduc) [reply packages/base/src/mainview.tsx:135]: Yes

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @martinRenou, great work!

**Pull request merged**